### ISSUE TITLE
Allow replacing fast-json-stringify with another output serializer

### ISSUE THREAD

**Issue by ilaif** [fastify/fastify#1934]
## 🚀 Feature Proposal

Allow replacing `fast-json-stringify` with another output serializer through a Fastify API. Similar to `fastify.setSchemaCompiler` we can have `fastify.setResponseSerializer`.

## Motivation

1. Need to wait for Fastify release to get a newer version of `fastify-json-stringify`.
2. I cannot use another serializer. For example, since I have a few added keywords to my `Ajv` instance, I would like to use Ajv to validate the responses before stringifying it, to apply the added keywords in the response as well.

## Example

```javascript
const Fastify = require('fastify')

const fastify = Fastify()

fastify.setResponseSerializer((statusCodeDefinition, externalSchema) => {
  return myCustomSerializer(statusCodeDefinition, externalSchema)
})
```

That is, inject the serializer instead of the default one here: https://github.com/fastify/fastify/blob/4cf5c338edcdece7272f9935da91e6b1080eb190/lib/validation.js#L13

Thanks, Fastify is the best!

**Comment by mcollina** [comment]
Let's go through things:

1. You can use package alias https://github.com/npm/rfcs/blob/latest/implemented/0001-package-aliases.md that landed in npm 6.9.0 (unfortunately this is not well documented in npm). Yarn supports them as well.

2. I would recommend against validating your responses before sending them. It's a lot of CPU work to validate a response, and you do not need it to protect against user input. However, I would agree that it'd be useful in development. 

If you happen to need a custom serializer, you might also check https://www.fastify.io/docs/latest/Reply/#serializerfunc.

I'm not opposed to this change anyway, as long as it does not slow us down in benchmark.

**Comment by Eomm** [comment]
I think we can archive this feature for v3

I'm working in a refact of the validation and serialization

**Comment by ilaif** [comment]
Cool, sounds good. LMK if I can be of help

**Comment by Eomm** [comment]
This has been implemented in #2023 and it will be released in v3

**Comment by Eomm** [comment]
Released in v3!
